src/unlock/omni_lock.rs

```diff
@@ -0,0 +1,157 @@
+use std::fmt::Display;
+
+use crate::{types::omni_lock::OmniLockWitnessLock, util::blake160};
+use ckb_types::{
+    bytes::{BufMut, Bytes, BytesMut},
+    packed::WitnessArgs,
+    prelude::*,
+};
+
+use ckb_crypto::secp::Pubkey;
+pub use ckb_types::prelude::Pack;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum IdentityFlags {
+    PubkeyHash = 0,
+}
+
+impl From<IdentityFlags> for u8 {
+    fn from(val: IdentityFlags) -> Self {
+        val as u8
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Identity {
+    pub flags: IdentityFlags,
+    pub blake160: Bytes,
```

Comment:
All `blake160` field please use `H160` type to enforce the length of the data.

Reply:
good idea